store/store_test.go

```diff
@@ -335,6 +340,58 @@ func (t *remoteRepoTestSuite) TestDownloadSyncFails(c *C) {
 	c.Assert(osutil.FileExists(tmpfile.Name()), Equals, false)
 }
 
+func (t *remoteRepoTestSuite) TestActualDownload(c *C) {
```

Comment:
Would be nice to have a test table here that specifies the variants, to avoid the three duplicated tests.